Subject: [PATCH] feature: 支持debian下的版本
---
Index: packages.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages.gradle b/packages.gradle
--- a/packages.gradle	(revision 992e26eee5c21d135e0b414a8b07caedc3e7b171)
+++ b/packages.gradle	(date 1752052414805)
@@ -426,12 +426,12 @@
             into "$DEB_BLD_DIR/debian"
         }
         def baseCommonDir = "$BASE_DIR/bigtop-packages/src/common/$NAME"
-        def versionCommonDir = "$baseCommonDir/${BASE_VERSION}"
+        def versionCommonDir = "$baseCommonDir/${PKG_VERSION}"
         // 1. 拷贝通用 debian 资源（不包括版本目录）
         copy {
             from fileTree(baseCommonDir) {
                 include '**/*'
-                exclude "${BASE_VERSION}/**"
+                exclude "${PKG_VERSION}/**"
             }
             into "$DEB_BLD_DIR/debian"
         }
Subject: [PATCH] feature: 取消bigtop-select 只能打rpm的限制
---
Index: bigtop.bom
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop.bom b/bigtop.bom
--- a/bigtop.bom	(revision b3084df291081144b974733eb28455f812311436)
+++ b/bigtop.bom	(date 1752216483054)
@@ -326,7 +326,7 @@
             relNotes = "Service package for Apache Bigtop information to use Ambari"
             version { base = bigtop.version; pkg = base - "-SNAPSHOT"; release = 1 }
             tarball { destination = "bigtop-select-${version.base}.tar.gz" }
-            packaging = 'rpm'
+//            packaging = 'rpm'
         }

         'bigtop-jsvc' {
