Subject: [PATCH] feature: 支持hadoop提前添加yarn包提升打包速度
---
Index: bigtop-packages/src/deb/hadoop/source/include-binaries
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop-packages/src/deb/hadoop/source/include-binaries b/bigtop-packages/src/deb/hadoop/source/include-binaries
new file mode 100644
--- /dev/null	(date 1752223557788)
+++ b/bigtop-packages/src/deb/hadoop/source/include-binaries	(date 1752223557788)
@@ -0,0 +1,1 @@
+debian/yarn-ui-bower.tar.gz
\ No newline at end of file
Index: bigtop-packages/src/common/hadoop/do-component-build
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop-packages/src/common/hadoop/do-component-build b/bigtop-packages/src/common/hadoop/do-component-build
--- a/bigtop-packages/src/common/hadoop/do-component-build	(revision 69294345caf3dd7a267bf0f6786462ada0fe8789)
+++ b/bigtop-packages/src/common/hadoop/do-component-build	(date 1752224196258)
@@ -132,7 +132,15 @@
 . $(dirname ${0})/maven_deploy.sh

 # bower_components copy here
-tar -zxvf ../../SOURCES/yarn-ui-bower.tar.gz -C ./hadoop-yarn-project/hadoop-yarn/hadoop-yarn-ui/src/main/webapp
+if [ -f /etc/redhat-release ]; then
+    echo "检测到 RedHat/CentOS，使用 RPM 源 yarn-ui-bower.tar.gz"
+    tar -zxvf ../../SOURCES/yarn-ui-bower.tar.gz \
+        -C ./hadoop-yarn-project/hadoop-yarn/hadoop-yarn-ui/src/main/webapp
+else
+    echo "检测到 Debian/Ubuntu，使用 debian 目录下的 yarn-ui-bower.tar.gz"
+    tar -xf debian/yarn-ui-bower.tar.gz \
+        -C ./hadoop-yarn-project/hadoop-yarn/hadoop-yarn-ui/src/main/webapp
+fi

 # rm -rf hadoop-yarn-ui  yarn lock
 rm -rf ./hadoop-yarn-project/hadoop-yarn/hadoop-yarn-ui/src/main/webapp/yarn.lock
